server/bot/commands/trivia.ts

```diff
@@ -0,0 +1,301 @@
+  timer: number
+  timeout: number
+  count: number
+  tempDB: DB
+  client: Client
+
+  constructor (triviaList: {question: string, answers: string[]}[], message: Message, botPlays: boolean, timeLimit: number, maxScore: number, revealAnswers: boolean, tempDB: DB, client: Client) {
+    this.channel = message.channel
+    this.author = message.author
+    this.message = message
+    this.questionList = triviaList
+    this.status = 'new question'
+    this.settings = { maxScore: maxScore, timeout: 120000, delay: timeLimit, botPlays: botPlays, revealAnswer: revealAnswers }
+    this.timer = null
+    this.count = 0
+    this.timeout = Date.now()
+    this.tempDB = tempDB
+    this.client = client
+  }
+
+  async sendScores () {
+    const currentScores = Object.values(this.scores).sort((a: number, b: number) => b - a)
+    const member = this.message.member.guild.members.get(this.client.user.id)
+    const color = member ? (member.roles.map(i => member.guild.roles.get(i)).sort(
+      (a, b) => a.position > b.position ? -1 : 1
+    ).find(i => i.color !== 0) || { color: 0 }).color : 0
+    const embed: EmbedOptions = {
+      title: 'Scores',
+      color,
+      timestamp: new Date().toISOString(),
+      fields: []
+    }
+    currentScores.forEach(score => {
+      embed.fields.push({
+        name: Object.keys(this.scores).find(key => this.scores[key] === score),
+        value: String(score),
+        inline: true
+      })
+    })
+    return { embed }
+  }
+
+  async stopTrivia () {
+    this.status = 'stop'
+    delete this.tempDB.trivia[this.channel.id]
+    this.settings = {
+      maxScore: 30,
+      timeout: 120000,
+      delay: 15000,
+      botPlays: false,
+      revealAnswer: true
+    }
+    this.scores = {}
+  }
+
+  async endGame () {
+    this.status = 'stop'
+    delete this.tempDB.trivia[this.channel.id]
+    if (this.scores) { this.channel.createMessage(await this.sendScores()) }
+    this.settings = {
+      maxScore: 30,
+      timeout: 120000,
+      delay: 15000,
+      botPlays: false,
+      revealAnswer: true
+    }
+    this.scores = {}
+  }
+
+  async newQuestion () {
+    for (let i of Object.values(this.scores)) {
+      if (i === this.settings.maxScore) {
+        this.endGame()
+        return true
+      }
+    }
+    if (!this.questionList) {
+      this.endGame()
+      return true
+    }
+    this.currentLine = this.questionList[Math.floor(Math.random() * this.questionList.length)]
+    this.questionList.splice(this.questionList.indexOf(this.currentLine), 1)
+    this.status = 'waiting for answer'
+    this.count += 1
+    this.timer = Date.now()
+    this.channel.createMessage(`**Question number ${this.count}!**\n\n${this.currentLine.question}`)
+
+    while (this.status !== 'correct answer' && (Date.now() - this.timer) <= this.settings.delay) {
+      if (Date.now() - this.timeout >= this.settings.timeout) {
+        const msg = `If you ${getInsult}s aren't going to play then I might as well stop.`
```

Comment:
`getInsult` is a function.
```suggestion
        const msg = `If you ${getInsult()}s aren't going to play then I might as well stop.`
```